packages/blade/src/tokens/theme/createTheme.ts

```diff
@@ -0,0 +1,376 @@
+              default: brandColors[400],
+              hover: brandColors[400],
+              focus: brandColors[400],
+              active: brandColors[400],
+            },
+          },
+          text: {
+            primary: {
+              default: foregroundOnBrandColorBankingDark,
+              hover: foregroundOnBrandColorBankingDark,
+              focus: foregroundOnBrandColorBankingDark,
+              active: foregroundOnBrandColorBankingDark,
+            },
+            secondary: {
+              default: foregroundOnSurfaceBankingDark,
+              hover: foregroundOnSurfaceBankingDark,
+              focus: foregroundOnSurfaceBankingDark,
+              active: foregroundOnSurfaceBankingDark,
+            },
+          },
+          icon: {
+            primary: {
+              default: foregroundOnBrandColorBankingDark,
+              hover: foregroundOnBrandColorBankingDark,
+              focus: foregroundOnBrandColorBankingDark,
+              active: foregroundOnBrandColorBankingDark,
+            },
+            secondary: {
+              default: foregroundOnSurfaceBankingDark,
+              hover: foregroundOnSurfaceBankingDark,
+              focus: foregroundOnSurfaceBankingDark,
+              active: foregroundOnSurfaceBankingDark,
+            },
+          },
+        },
+      },
+    },
+  };
+
+  return overrideTheme({
+    baseThemeTokens: bankingTheme,
+    overrides: darkThemeBankingOverrides,
+  });
+};
+
+/**
+ * @param {Object} themeConfig - The brand color and overrides to apply to the theme
+ * @param {string} themeConfig.brandColor - The brand color to use to generate the theme. Can be in hex, rgb, or hsl format.
+ * @description
+ * Creates a Blade Theme based on the custom brand color and overrides passed in
+ * @returns The Theme Tokens with the custom brand colors and overrides applied
+ * @example
+ * const theme = createTheme({ brandColor: '#19BEA2', overrides: {} })
+ **/
+export const createTheme = ({
+  brandColor,
+  overrides: extraOverrides = {},
+}: {
+  brandColor: ColorInput;
+  overrides?: DeepPartial<ThemeTokens>;
+}): ThemeTokens => {
+  const chromaticBrandColors = generateChromaticBrandColors(brandColor);
+  // Get theme tokens for overriding onLight colors which are based on payment theme
+  const brandedLightTheme = getLightTheme(chromaticBrandColors);
+  // Get theme tokens for overriding onDark colors which are based on banking theme
+  const brandedDarkTheme = getDarkTheme(chromaticBrandColors);
+
+  // Merge theme with light colors from payment theme and dark colors from banking theme
+  const brandedThemeTokens = overrideTheme({
+    baseThemeTokens: paymentTheme,
+    overrides: {
+      colors: {
+        onLight: {
+          ...brandedLightTheme.colors.onLight,
+        },
+        onDark: {
+          ...brandedDarkTheme.colors.onDark,
+        },
+      },
+      ...extraOverrides, // Apply any extra overrides passed in by the consumer
```

Comment:
so what happens when i pass


```js


const extraOverrides = {
  onLight: {
    action: {
      background: {
        primary: {
          default: '#ffad',
          hover: '#ffad',
          focus: '#ffae',
          active: '#ffah',
        },
      }
    }
  }
}

overrides: {
  colors: {
    onLight: {
      ...brandedLightTheme.colors.onLight,
    },
    onDark: {
      ...brandedDarkTheme.colors.onDark,
    },
  },
  ...extraOverrides,
}
```


because since we are spreading the object as a consumer - I've just passed brand color and some overrides but since we are spreading `extraOverrides` we'll override only my overrides but then where will the other colors come from? `..brandedLightTheme.colors.onLight,` will not have any effect then?

Reply:
Resolved with latest commits